linera-views/src/views/bucket_queue_view.rs

```diff
@@ -0,0 +1,700 @@
+                    Some(vec) => vec,
+                    None => {
+                        let value = values[pos].as_ref().ok_or(ViewError::MissingEntries)?;
+                        pos += 1;
+                        &bcs::from_bytes::<Vec<T>>(value)?
+                    }
+                };
+                let end = if count_remain <= size {
+                    position + count_remain
+                } else {
+                    position + size
+                };
+                for element in &vec[position..end] {
+                    elements.push(element.clone());
+                }
+                if size >= count_remain {
+                    count_remain = 0;
+                    break;
+                }
+                count_remain -= size;
+                position = 0;
+            }
+        }
+        if count_remain > 0 {
+            let count_read = std::cmp::min(count_remain, self.new_back_values.len());
+            elements.extend(self.new_back_values.range(0..count_read).cloned());
+        }
+        Ok(elements)
+    }
+
+    /// Returns the last element of a bucket queue view
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::context::create_test_memory_context;
+    /// # use linera_views::bucket_queue_view::BucketQueueView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_memory_context();
+    /// let mut queue = BucketQueueView::<_, u128, 5>::load(context).await.unwrap();
+    /// queue.push_back(34);
+    /// queue.push_back(37);
+    /// queue.push_back(47);
+    /// assert_eq!(queue.read_front(2).await.unwrap(), vec![34, 37]);
+    /// # })
+    /// ```
+    pub async fn read_front(&self, count: usize) -> Result<Vec<T>, ViewError> {
+        self.read_context(self.cursor.position, count).await
+    }
+
+    /// Returns the last element of a bucket queue view
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::context::create_test_memory_context;
+    /// # use linera_views::bucket_queue_view::BucketQueueView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_memory_context();
+    /// let mut queue = BucketQueueView::<_, u128, 5>::load(context).await.unwrap();
+    /// queue.push_back(34);
+    /// queue.push_back(37);
+    /// queue.push_back(47);
+    /// assert_eq!(queue.read_front(2).await.unwrap(), vec![34, 37]);
+    /// # })
+    /// ```
+    pub async fn read_back(&self, count: usize) -> Result<Vec<T>, ViewError> {
+        if count <= self.new_back_values.len() {
+            let start = self.new_back_values.len() - count;
+            Ok(self
+                .new_back_values
+                .range(start..)
+                .cloned()
+                .collect::<Vec<_>>())
+        } else {
+            let mut increment = self.count() - count;
+            let Some((i_block, mut position)) = self.cursor.position else {
+                unreachable!();
```

Comment:
If the unreachable were called, then it would mean that the `cursor.position = None`. In that case, you would get that `self.count() == self.new_back_values.len()`. But that case would be covered by the first if loop.